sos/plugins/openstack_aodh.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2017 Red Hat, Inc., Sachin Patil <psachin@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class OpenStackAodh(Plugin, RedHatPlugin):
+    """OpenStack Alarm service"""
+    plugin_name = "openstack_aodh"
+    profiles = ('openstack', 'openstack_controller')
+
+    packages = (
+        'openstack-aodh-api',
+        'openstack-aodh-listener',
+        'openstack-aodh-notifier',
+        'openstack-aodh-evaluator,'
+        'openstack-aodh-common')
+
+    requires_root = False
+
+    def setup(self):
+        self.add_copy_spec("/etc/aodh/")
+
+        self.limit = self.get_option("log_size")
+        if self.get_option("all_logs"):
+            self.add_copy_spec("/var/log/aodh/",
+                               sizelimit=self.limit)
+        else:
+            self.add_copy_spec("/var/log/aodh/*.log",
+                               sizelimit=self.limit)
+
+        vars_all = [p in os.environ for p in [
+            'OS_USERNAME', 'OS_PASSWORD', 'OS_AUTH_TYPE']]
+
```

Comment:
I'm not a fan of this `[...[\nblah]]` style, but I think we let another one into the tree recently, so I'm happy to let this slide and then fix it up later.

For reference, what I'd prefer is either:

- Assign the list of names to check to a variable (so it all fits onto one line)
- If you have to do it in an open list comprehension, use the `sos` multi-line list convention:

```python
    vars_all = [p in os.environ for p in [
        "FOO", "BAR", "BAZ", "QUX"
    ]]
````

It draws attention to the different parts of the statement better and makes bracketing errors when modifying the lists less likely.